### PR TITLE
Export/339_345 pricing module export integration and CSVProvider implement

### PR DESCRIPTION
**Pull request by akak1977**

### CODE REVIEW

File: Modules/vc-module-export/VirtoCommerce.ExportModule.Data/Services/CsvExportProvider.cs [hunk 1]
```diff
@@ -1,31 +1,123 @@
...
 
 namespace VirtoCommerce.ExportModule.Data.Services
 {
-    public class CsvExportProvider : IExportProvider
+    public sealed class CsvExportProvider : IExportProvider
     {
-        private readonly Stream _stream;
+        private static class DynamicMethodGenerator
+        {
+            public delegate object GetMethodDelegate(object source);
```
Comment (yecli) [inline Modules/vc-module-export/VirtoCommerce.ExportModule.Data/Services/CsvExportProvider.cs]: Maybe we could make interface like: 
```
        public interface IPropertyValueGetter
        {
            object GetPropertyValue(object obj, MemberInfo memberInfo);
        }
```
implement it with ILPropertyValueGetter and incapsulate property value get logic and caching there?

File: Modules/vc-module-export/VirtoCommerce.ExportModule.Data/Services/CsvExportProvider.cs [hunk 2]
```diff
@@ -1,31 +1,123 @@
...
+                _streamWriter.Write(@"""");
+                bFirst = false;
+            }
+            _streamWriter.WriteLine();
         }
 
         public void WriteRecord(object objectToRecord)
         {
-            throw new System.NotImplementedException();
+            var bFirst = true;
```
Comment (yecli) [inline Modules/vc-module-export/VirtoCommerce.ExportModule.Data/Services/CsvExportProvider.cs]: Please name without type, e.g. first

Comment (akak1977) [reply Modules/vc-module-export/VirtoCommerce.ExportModule.Data/Services/CsvExportProvider.cs]: OK

File: Modules/vc-module-export/VirtoCommerce.ExportModule.Data/Services/CsvExportProvider.cs (hunk 1 above)
Comment (akak1977) [reply Modules/vc-module-export/VirtoCommerce.ExportModule.Data/Services/CsvExportProvider.cs]: Not immediately, I think. Let's look how it works. May be later.

**Pull request merged**